source/Ship.cpp

```diff
@@ -2175,7 +2175,7 @@ shared_ptr<Ship> Ship::Board(bool autoPlunder)
 		SetShipToAssist(shared_ptr<Ship>());
 		SetTargetShip(shared_ptr<Ship>());
 		bool helped = victim->isDisabled;
-		victim->hull = max(victim->hull, victim->MinimumHull());
+		victim->hull = max(victim->hull, victim->MinimumHull() * 1.5);
```

Comment:
Should this be capped at the maximum hull value as well? We don't want ships with more than 67% minimum hull to end up with more hull than they started with...

Reply:
During Ship::DoGeneration there exists a check to prevent the hull (or shields) of a ship from being above the maximum, so it seemed redundant to me to also clamp the hull to below the maximum value here.
https://github.com/endless-sky/endless-sky/blob/1af961037f9133fe7a4b8fdce3033b2173db2025/source/Ship.cpp#L2015-L2018

Reply:
It's likely appropriate to add the clamp both here and in `TakeDamage`, to preserve the class invariant externally (that is, a ship cannot ever have more "X" than "MaxX").

If no other class could possibly observe or influence X between when it is above MaxX and when it gets clamped within `MinX <= X <= MaxX`, then it's fine. Generally that is only true within a method call (and doesn't hold in general if exceptions are possible).